runelite-client/src/main/java/net/runelite/client/plugins/combatlevel/CombatLevelOverlay.java

```diff
@@ -106,27 +106,36 @@ private String getLevelsUntilTooltip()
 
 		// create tooltip string
 		StringBuilder sb = new StringBuilder();
-		sb.append(ColorUtil.wrapWithColorTag("Next combat level:</br>", COMBAT_LEVEL_COLOUR));
+		boolean isMeleeCombatDominating = ((attackLevel + strengthLevel) > (Experience.MAX_REAL_LEVEL / 2 + Experience.MAX_REAL_LEVEL));
 
-		if ((attackLevel + strengthLevel + meleeNeed) <= Experience.MAX_REAL_LEVEL * 2)
+		if (client.getLocalPlayer().getCombatLevel() >= Experience.MAX_COMBAT_LEVEL)
 		{
-			sb.append(meleeNeed).append(" Attack/Strength</br>");
+			sb.append(ColorUtil.wrapWithColorTag("Max combat level achieved", COMBAT_LEVEL_COLOUR));
 		}
-		if ((hitpointsLevel + defenceLevel + hpDefNeed) <= Experience.MAX_REAL_LEVEL * 2)
+		else
 		{
-			sb.append(hpDefNeed).append(" Defence/Hitpoints</br>");
-		}
-		if ((rangeLevel + rangeNeed) <= Experience.MAX_REAL_LEVEL)
-		{
-			sb.append(rangeNeed).append(" Ranged</br>");
-		}
-		if ((magicLevel + magicNeed) <= Experience.MAX_REAL_LEVEL)
-		{
-			sb.append(magicNeed).append(" Magic</br>");
-		}
-		if ((prayerLevel + prayerNeed) <= Experience.MAX_REAL_LEVEL)
-		{
-			sb.append(prayerNeed).append(" Prayer");
+			sb.append(ColorUtil.wrapWithColorTag("Next combat level:</br>", COMBAT_LEVEL_COLOUR));
+
+			if ((attackLevel + strengthLevel) < Experience.MAX_REAL_LEVEL * 2)
+			{
+				sb.append(meleeNeed).append(" Attack/Strength</br>");
+			}
+			if ((hitpointsLevel + defenceLevel) < Experience.MAX_REAL_LEVEL * 2)
+			{
+				sb.append(hpDefNeed).append(" Defence/Hitpoints</br>");
+			}
+			if (rangeLevel < Experience.MAX_REAL_LEVEL && !isMeleeCombatDominating)
```

Comment:
I don't understand why the check for `!isMeleeCombatDominating` is added? Even with the combat level being computed by the "melee combat level", that doesn't mean you cannot gain combat levels from ranged levels. It just means that you have to gain ranged levels until `range * 3/2 > attack+str`